src/exo/LoopIR_scheduling.py

```diff
@@ -1914,10 +1881,20 @@ def map_s(self, sc):
         if s is self.alloc_stmt:
             if self.n_lifts > len(self.ctrl_ctxt):
                 raise SchedulingError(
-                    "specified lift level {self.n_lifts} "
-                    + "is higher than the number of loop "
-                    + "{len(self.ctrl_ctxt)}"
+                    f"specified lift level {self.n_lifts} "
+                    f"is more than {len(self.ctrl_ctxt)}, "
+                    f"the number of loops "
+                    f"and ifs above the allocation"
                 )
+            if len(s.type.shape()) > 0:
```

Comment:
Don't test `len`. Check truthiness directly.
```suggestion
            if s.type.shape():
```